providers/nomad/helpers.go

```diff
@@ -0,0 +1,297 @@
+	jobStatus := *job.Status
+	jobCreatedAt := *job.SubmitTime
+	podCondition := convertJobStatusToPodCondition(jobStatus)
+	containerStatusesMap := allocToContainerStatuses(allocs)
+
+	// containerPorts are specified for task in a task
+	// group
+	var containerPorts []v1.ContainerPort
+	for _, tg := range job.TaskGroups {
+		for _, task := range tg.Tasks {
+			for _, taskNetwork := range task.Resources.Networks {
+				for _, dynamicPort := range taskNetwork.DynamicPorts {
+					// TODO: Dynamic ports aren't being reported via the
+					// Nomad `/jobs` endpoint.
+					containerPorts = append(containerPorts, v1.ContainerPort{
+						Name:     dynamicPort.Label,
+						HostPort: int32(dynamicPort.Value),
+						HostIP:   taskNetwork.IP,
+					})
+				}
+			}
+
+			containers = append(containers, v1.Container{
+				Name:    task.Name,
+				Image:   fmt.Sprintf("%s", task.Config["image"]),
+				Command: strings.Split(fmt.Sprintf("%s", task.Config["command"]), ""),
+				Args:    strings.Split(fmt.Sprintf("%s", task.Config["args"]), " "),
+				Ports:   containerPorts,
+			})
+
+			containerStatus := containerStatusesMap[task.Name]
+			containerStatus.Image = fmt.Sprintf("%s", task.Config["image"])
+			containerStatus.ImageID = fmt.Sprintf("%s", task.Config["image"])
+			containerStatues = append(containerStatues, containerStatus)
+		}
+	}
+
+	pod := v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              *job.Name,
+			Namespace:         "default",
+			CreationTimestamp: metav1.NewTime(time.Unix(jobCreatedAt, 0)),
+		},
+		Spec: v1.PodSpec{
+			NodeName:   p.nodeName,
+			Volumes:    []v1.Volume{},
+			Containers: containers,
+		},
+		Status: v1.PodStatus{
+			Phase:             jobStatusToPodPhase(jobStatus),
+			Conditions:        []v1.PodCondition{podCondition},
+			Message:           "",
+			Reason:            "",
+			HostIP:            "", // TODO: find out the HostIP
+			PodIP:             "", // TODO: find out the equalent for PodIP
+			ContainerStatuses: containerStatues,
+		},
+	}
+
+	return &pod, nil
+}
+
+func allocToContainerStatuses(allocs []*nomad.AllocationListStub) map[string]v1.ContainerStatus {
+	containerStatusesMap := map[string]v1.ContainerStatus{}
+
+	for _, alloc := range allocs {
+		for name, taskState := range alloc.TaskStates {
+			containerState, readyFlag := convertTaskStateToContainerState(taskState.State,
+				taskState.StartedAt,
+				taskState.FinishedAt,
+			)
+			containerStatusesMap[name] = v1.ContainerStatus{
+				Name:         name,
+				RestartCount: int32(taskState.Restarts),
+				Ready:        readyFlag,
+				State:        containerState,
+			}
+		}
+	}
+
+	return containerStatusesMap
+}
+
+func jobStatusToPodPhase(status string) v1.PodPhase {
+	switch status {
+	case "pending":
+		return v1.PodPending
+	case "running":
+		return v1.PodRunning
+	// TODO: Make sure we take PodFailed into account.
+	case "dead":
```

Comment:
just curious: do you have a "PodSucceeded" state?

Reply:
Currently, we don't have an equivalent state to "PodSucceeded". I will try taking a peak again if I could use any other state from the job spec.